services/player-service/application/src/main/java/com/kjeldsen/player/application/usecases/GenerateSingleTrainingUseCase.java

```diff
@@ -27,26 +29,59 @@ public void generate(PlayerId playerId, List<PlayerSkill> skills, Integer days)
     }
 
     private void generateAndStoreEvent(PlayerId playerId, PlayerSkill playerSkill, int currentDay) {
-        PlayerTrainedEvent.builder()
+        PlayerTrainingEvent playerTrainingEvent = PlayerTrainingEvent.builder()
             .eventId(EventId.generate())
             .eventDate(Instant.now())
             .playerId(playerId)
             .skill(playerSkill)
-            .points(generatePoints(currentDay))
+            .points(generateRise(currentDay))
             .build();
+        playerTrainingEventWriteRepository.save(playerTrainingEvent);
 
         // TODO y el evento PlayerTrainedEvent se lo mandas a un repository de tipo write (mira ejemplos hechos) y lo guardas.
         // luego te metes a la base de datos y compruebas que este ahi
         // cuando este ahi, buscas CQRS y lees un poco sobre el tema y te ves un video. PAra que entiendas por que has usado un reposiotrio al que
         // llamas write (veras que hay read tmb) :D
+
     }
 
-    private int generatePoints(int currentDay) {
-        // TODO dali mandale ahi
-        if (currentDay < 2) {
+    private int generateRise(int currentDay) {
+
+        double randomProbabilityRise = Math.random();
+        float probability = 1 / 14;
+        double probabilityRaise = probability * currentDay; // Sumatorio de la prob diaria por el numero de dias que lleva
+        int risePoints = 0;
+
+        if (randomProbabilityRise <= probabilityRaise) {
+            risePoints = generatePoints();
+        }
+
+        return risePoints;
+    }
+
+    private int generatePoints() {
```

Comment:
lets have a module called engine in the project and we move this kind of calculations there. Business uses the language engine so let's stick to it in code as well